File: sim-bot-desktop-pack-out/sim-bot-app/scripts/weixin-process.mjs
/**
 * 检测 / 结束本机微信（Weixin.exe）主进程。仅 Windows。
 */
import { execFileSync } from 'child_process';

/**
 * @returns {number | null} 运行中微信进程 PID；未找到则 null
 */
export function getRunningWeixinPid() {
  if (process.platform !== 'win32') return null;
  try {
    const out = execFileSync(
      'powershell.exe',
      [
        '-NoProfile',
        '-NonInteractive',
        '-Command',
        '$p = Get-Process -Name Weixin -ErrorAction SilentlyContinue | Sort-Object Id | Select-Object -First 1; if ($p) { $p.Id }',
      ],
      { encoding: 'utf8', windowsHide: true, maxBuffer: 64 * 1024 }
    );
    const n = Number(String(out).trim());
    return Number.isFinite(n) && n > 0 ? n : null;
  } catch {
    return null;
  }
}

function sleepMs(ms) {
  try {
    execFileSync(
      'powershell.exe',
      ['-NoProfile', '-NonInteractive', '-Command', `Start-Sleep -Milliseconds ${Math.max(0, ms | 0)}`],
      { windowsHide: true, timeout: Math.max(5000, ms + 2000) }
    );
  } catch {
    /* ignore */
  }
}

/**
 * 结束所有 Weixin.exe（含子进程树）。用于桌面启动前干净注入。
 * @returns {boolean} 结束时是否已无微信进程
 */
export function stopRunningWeixin({ waitMs = 2500 } = {}) {
  if (process.platform !== 'win32') return true;
  if (!getRunningWeixinPid()) return true;
  try {
    execFileSync('taskkill', ['/IM', 'Weixin.exe', '/F', '/T'], {
      windowsHide: true,
      timeout: 15000,
    });
  } catch {
    return !getRunningWeixinPid();
  }
  const deadline = Date.now() + waitMs;
  while (Date.now() < deadline) {
    if (!getRunningWeixinPid()) return true;
    sleepMs(300);
  }
  return !getRunningWeixinPid();
}

/**
 * 仅桌面「首次启动注入」时允许先关微信；看门狗重试不得带此标记。
 * 需同时 WECHAT_BOOTSTRAP_INJECT=1，且 WECHAT_QUIT_BEFORE_INJECT≠0。
 */
export function shouldQuitWeixinBeforeInject() {
  return (
    process.env.WECHAT_BOOTSTRAP_INJECT === '1' &&
    process.env.WECHAT_QUIT_BEFORE_INJECT !== '0'
  );
}

File: utils/weixin_process.py
"""Windows 微信进程探测"""
from __future__ import annotations

import os
import subprocess
import time
from pathlib import Path

_WEIXIN_IMAGE_NAMES = ('Weixin.exe', 'WeChat.exe')


def _tasklist_pids(image_name: str) -> list[int]:
    if os.name != 'nt':
        return []
    try:
        out = subprocess.check_output(
            ['tasklist', '/FI', f'IMAGENAME eq {image_name}', '/FO', 'CSV', '/NH'],
            encoding='utf-8',
            errors='ignore',
            creationflags=getattr(subprocess, 'CREATE_NO_WINDOW', 0),
        )
    except Exception:
        return []
    pids = []
    for line in out.splitlines():
        line = line.strip()
        if not line or image_name.lower() not in line.lower():
            continue
        parts = [p.strip('"') for p in line.split('","')]
        if len(parts) >= 2 and parts[0].lower() == image_name.lower():
            try:
                pids.append(int(parts[1]))
            except ValueError:
                continue
    return pids


def get_running_weixin_pid() -> int | None:
    for name in _WEIXIN_IMAGE_NAMES:
        pids = _tasklist_pids(name)
        if pids:
            return pids[0]
    return None


def get_running_weixin_executable() -> str | None:
    """从正在运行的微信进程读取 exe 路径（冷注入关进程前应先调用）。"""
    pid = get_running_weixin_pid()
    if not pid:
        return None
    try:
        out = subprocess.check_output(
            [
                'powershell',
                '-NoProfile',
                '-Command',
                f'(Get-Process -Id {int(pid)} -ErrorAction SilentlyContinue).Path',
            ],
            encoding='utf-8',
            errors='ignore',
            creationflags=getattr(subprocess, 'CREATE_NO_WINDOW', 0),
            timeout=8,
        ).strip()
        if out and Path(out).is_file():
            return str(Path(out).resolve())
    except Exception:
        pass
    return None


def stop_running_weixin(wait_ms: int = 3000) -> bool:
    if os.name != 'nt':
        return False
    try:
        for name in _WEIXIN_IMAGE_NAMES:
            subprocess.run(
                ['taskkill', '/IM', name, '/F', '/T'],
                check=False,
                creationflags=getattr(subprocess, 'CREATE_NO_WINDOW', 0),
            )
        time.sleep(wait_ms / 1000.0)
        return get_running_weixin_pid() is None
    except Exception:
        return False


def should_quit_weixin_before_inject() -> bool:
    return os.getenv('WECHAT_QUIT_BEFORE_INJECT', '0') == '1'
